### ISSUE TITLE
fix(app): Streamlitアプリの状態ファイル（photon_app_state.json）が学習終了後も running のまま更新されない

### ISSUE THREAD

**Issue by Kewton** [Kewton/photon-mlx#59]
## 現象

`app/photon_app.py` で起動した学習ジョブが完了・失敗しても、`.cache/photon_app_state.json` の状態が更新されず、`status=running, last_step=0, val_loss=0.0` のまま残る。

### 実例

```json
{
  "train_20260420_175030": {
    "status": "running",      // ← 実際はプロセス終了済み
    "last_step": 0,           // ← 実際は 1000 step 完了
    "val_loss": 0.0,          // ← 実際は best 1.6239
    "max_steps": 1000
  }
}
```

学習プロセス（PID 33403）は既に終了しており、ログ（`logs/train_20260420_175030.log`）には `Training complete. Final loss: 0.0219` が記録されているが、状態ファイルには反映されていない。

## 原因

`app/photon_app.py` の状態同期（`_is_process_running` チェック）は **Streamlit の UI レンダリング時にのみ実行**される。

```python
# photon_app.py:278-280 — UIレンダリング中にのみ動作
running = _is_process_running(job.pid)
if job.status == "running" and not running:
    job.status = "completed"
```

このため以下のケースで状態ファイルが古いまま残る：

1. **学習中にブラウザタブを閉じた場合** — 状態同期コードが実行されない
2. **Streamlit プロセスを停止した場合** — アプリ自体が動いていない
3. **学習ジョブが長時間実行され、途中でブラウザを閉じた場合** — 完了時点で反映されない

## 影響

- 状態ファイルを外部ツール（CLI スクリプト、モニタリングなど）から参照しても不正確な情報が返る
- アプリを再起動した時の表示が実態と乖離する
- 学習の進捗・結果を確実に把握できない

### 案A：バックグラウンドポーリング（推奨）

Streamlit の `st.session_state` と別に、**バックグラウンドスレッドでジョブ状態を定期ポーリング**して状態ファイルを更新する。

```python
# app/photon_app.py
import threading

def _background_sync_worker():
    while True:
        state = _load_state()
        updated = False
        for job in state.training_jobs.values():
            if job.status == "running":
                running = _is_process_running(job.pid)
                progress = _parse_log(job.log_file)
                if not running:
                    job.status = "completed" if progress["last_step"] > 0 else "failed"
                    updated = True
                elif progress["last_step"] > job.last_step:
                    job.last_step = progress["last_step"]
                    job.val_loss = progress["val_loss"]
                    updated = True
        if updated:
            _save_state(state)
        time.sleep(30)  # 30秒ごとにポーリング

# アプリ起動時に1回だけ起動
if "sync_thread_started" not in st.session_state:
    threading.Thread(target=_background_sync_worker, daemon=True).start()
    st.session_state.sync_thread_started = True
```

### 案B：独立した同期プロセス（より堅牢）

`scripts/sync_app_state.py` を追加し、cron や launchd で定期実行する。Streamlit アプリとは独立して動作するため、アプリ停止時でも状態が正確に保たれる。

```python
# scripts/sync_app_state.py
# アプリのデータモデルをインポートして状態同期
# 30秒〜1分ごとに launchd / cron から実行
```

### 案C：ログファイル末尾から完了検知（最小実装）

`_parse_log()` を拡張して `"Training complete."` の行を検出したら `status=completed` に設定する。

```python
def _parse_log(log_path: str) -> dict:
    result = {...}
    ...
    if "Training complete" in content:
        result["completed"] = True
        result["final_loss"] = ...
    return result
```

## 関連ファイル

- `app/photon_app.py:126-150` — `_is_process_running()`, `_parse_log()`
- `app/photon_app.py:270-320` — training_jobs のステータス同期セクション
- `app/photon_app.py:430-470` — index_jobs のステータス同期セクション（同じ問題あり）

## 受入条件

- [ ] 学習プロセス終了後、状態ファイルの `status` が `completed` / `failed` に更新されること
- [ ] 学習中のステップ数・val_loss が（ブラウザを閉じていても）状態ファイルに反映されること
- [ ] Streamlit アプリを再起動しても状態が正確に表示されること
- [ ] index_jobs についても同様に状態同期が行われること
- [ ] 既存テストが全パスすること
- [ ] `ruff check` / `ruff format --check` がパスすること
